libs/header/p2p/exchange.go

```diff
@@ -236,12 +236,17 @@ func (ex *Exchange[H]) performRequest(
 		cancel()
 		switch err {
 		default:
+			reqErr = err
 			log.Debug(err)
 			continue
 		case context.Canceled, context.DeadlineExceeded, nil:
+			if err != nil && reqErr != nil {
+				err = reqErr
+			}
```

Comment:
can you document why you're doing this? isn't the whole point of this loop with trusted peers is so that if a request fails to one peer, you can move on to next peer and ask them and if it's successful, just return successfully? This makes is so if previous peer had error, you will still return that err even if the next peer was successful, no?

Also can you make the debug log on #L240 more descriptive ?

Reply:
Actually here we are handling context.Cancel error. If reqError is not empty, then it means that we received errors some errors from our trusted peers, and IMO it's better to return reqError(we can reqError.Join() after moving to Go1.20)

Reply:
Okay but still - look at what the caller if this is doing. If one request succeeds and some other one failed, we will still error out bc one failed.

Reply:
n/a